### PR TITLE
Updated doc pages

### PR DESCRIPTION
**Pull request by ashkan-saeedi-mazdeh**
Updated a few doc pages in step by step part.

Also according to issue:#1000 I added topics of getting started to the main page as the first step.

### CODE REVIEW

**Comment by ashkan-saeedi-mazdeh** [comment]
Referencing issue #1000  due to formatting issue above. Not used to markdown yet :(

**Comment by ashkan-saeedi-mazdeh** [comment]
Also updated the page for observers. We should move it to step by step tutorials later on as well. It's too advanced for getting started.

File: Step-by-step-Tutorials/A-Service-is-a-Collection-of-Communicating-Actors.md [hunk 1]
```diff
@@ -130,7 +130,7 @@ var e0 = GrainClient.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
 var e1 = GrainClient.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
 var e2 = GrainClient.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
 var e3 = GrainClient.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
-var e4 = GrainClient.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
+var e4 = GrainClient.GrainFactory.GrainFactory.GetGrain<IEmployee>(Guid.NewGuid());
```
Comment (gabikliot) [inline Step-by-step-Tutorials/A-Service-is-a-Collection-of-Communicating-Actors.md]: too much `GrainFactory`.

File: Step-by-step-Tutorials/A-Service-is-a-Collection-of-Communicating-Actors.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ public class Manager : Orleans.Grain, IManager
 {
     public override Task ActivateAsync()
     {
-        _me = GrainFactory.GetGrain<IEmployee>(this.GetPrimaryKey());
```
Comment (sergeybykov) [inline Step-by-step-Tutorials/A-Service-is-a-Collection-of-Communicating-Actors.md]: From within grain code it should be `this.GrainFactory.GetGrain<>` or `GrainFactory.GetGrain<>`, not `this.GetGrain<>` which won't compile.

File: Getting-Started-With-Orleans/Observers.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ title: Client Observers
 ---
 {% include JB/setup %}
 
+**NOte:** Support for observers might be removed in a future version if there is not demand for it since simple in memory streams can support the same concept with more power and flexibility.
```
Comment (sergeybykov) [inline Getting-Started-With-Orleans/Observers.md]: Typo - "Note"

File: Getting-Started-With-Orleans/Observers.md [hunk 4]
```diff
@@ -21,6 +23,97 @@ It then calls a static method on the observer factory, `CreateObjectReference()`
...
+
+``` csharp
+public interface IChat : IGrainObserver
+{
+    void ReceiveMessage(string message);
+}
+
+```
+
+The only special thing is that the interface should inherit from `IGrainObserver`. Now any client which wants to observe this messages should implement a class which inherit from `IChat`.
```
Comment (sergeybykov) [inline Getting-Started-With-Orleans/Observers.md]: "a class which implements `IChat`" will be more accurate I think.

**Comment by gabikliot** [comment]
I have merged and made some fixes directly in the gh-pages branch.

**Pull request merged**